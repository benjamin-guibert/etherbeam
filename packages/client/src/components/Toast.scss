.toast-header {
  background-color: transparent !important;
  color: inherit !important;
}
