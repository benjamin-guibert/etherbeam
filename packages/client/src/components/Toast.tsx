import React, { ReactElement, ReactNode, useState } from 'react'
import BsToast from 'react-bootstrap/Toast'
import './Toast.scss'

interface ToastProps {
  type: 'error'
  className?: string
  children: ReactNode
}

const Toast = ({ type, className = '', children }: ToastProps): ReactElement => {
  const getDataFromType = (): { variant: string; title: string } => {
    switch (type) {
      case 'error':
        return {
          title: 'Error',
          variant: 'bg-danger',
        }
      default:
        return {
          title: '',
          variant: '',
        }
    }
  }

  const [show, setShow] = useState<boolean>(true)

  const { variant, title } = getDataFromType()

  return (
    <BsToast className={[className, variant].join(' ')} onClose={() => setShow(false)} show={show} autohide>
      <BsToast.Header>
        <strong className="mr-auto">{title}</strong>
      </BsToast.Header>
      <BsToast.Body>{children}</BsToast.Body>
    </BsToast>
  )
}

export default Toast
