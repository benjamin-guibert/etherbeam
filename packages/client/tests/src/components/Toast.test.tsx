import * as React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import { act } from 'react-dom/test-utils'
import Toast from 'components/Toast'

describe('<Toast />', () => {
  beforeEach(() => {
    jest.clearAllTimers()
    jest.useFakeTimers()
  })

  let component: ReactWrapper

  describe('Error', () => {
    beforeAll(() => {
      component = mount(
        <Toast type="error" className="app-class">
          Message.
        </Toast>
      )
    })

    it('should render', () => {
      const toast = component.children().first()
      expect(toast.prop('show')).toBeTruthy()
      expect(toast.hasClass('bg-danger')).toBeTruthy()
      expect(toast.hasClass('app-class')).toBeTruthy()
      expect(toast.find('ToastHeader').contains('Error'))
      expect(toast.find('ToastBody').contains('Message.'))
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Auto-hide', () => {
    beforeAll(async () => {
      await act(async () => {
        component = mount(<Toast type="error">Message.</Toast>)

        await new Promise(setImmediate)
        jest.runTimersToTime(10 * 1000)
        await new Promise(setImmediate)
        component.update()
      })
    })

    it('should hide', () => {
      const toast = component.children().first()
      expect(toast.prop('show')).toBeFalsy()
    })
  })
})
