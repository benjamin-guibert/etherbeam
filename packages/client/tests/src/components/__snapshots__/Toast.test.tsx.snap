// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> Error should match snapshot 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="fade toast app-class bg-danger show"
  role="alert"
>
  <div
    class="toast-header"
  >
    <strong
      class="mr-auto"
    >
      Error
    </strong>
    <button
      class="close ml-2 mb-1"
      data-dismiss="toast"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
      <span
        class="sr-only"
      >
        Close
      </span>
    </button>
  </div>
  <div
    class="toast-body"
  >
    Message.
  </div>
</div>
`;
